### PR TITLE
Create Server Web Service

### PR DESCRIPTION
**Pull request by Gidoren**
Create a web service endpoint to wait for the public key of the client to spin up docker, create the keys, return the public keys of the server and configure the interface of the wireguard interface on the container.

### CODE REVIEW

**Comment by Gidoren** [comment]
Has been tested with Jons code. it is integrated with it provided I make an env file to pass the correct server ip so we do not compromise it on the public repo

File: vps-service/returnKey.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	//store in .env please
+	url := "localhost:3000"
+	fmt.Println("URL:>", url)
+	cmd := exec.Command("genkeys")
+	cmd.Stdout = os.Stdout
+	err := cmd.Run()
+	if err != nil {
+		log.Printf("Didnt generate keys")
+	}
+	// else {
```
Comment (ZeronTheXeon) [inline vps-service/returnKey.go]: Is this commented code needed?

File: vps-service/restPut.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	case "POST":
+		fmt.Println("Recieved POST request...")
+		fmt.Println("")
+		//Unmarshall the passed in JSON
+		body, _ := ioutil.ReadAll(r.Body)
+		var jsonValue map[string]interface{}
+		json.Unmarshal(body, &jsonValue)
+		publicKey := jsonValue["Public Key"].(string)
+		cmd := exec.Command("wg", "set", "wg0", "peer",
+			publicKey, "allowed-ips", "10.10.0.2/32,fd86:ea04:1111::2/128")
```
Comment (ZeronTheXeon) [inline vps-service/restPut.go]: If the allowedIPs are constant, can we extract them out to a global const?

File: vps-service/restPut.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+//creates a struct for json marshalling
+func sendIt() payload {
+	//replace with the connection manager ip and endpoing
+	//store in .env please
+	content, err := ioutil.ReadFile("public")
+	if err != nil {
+		log.Fatal(err)
+	}
+	public := string(content)
+	server := "10.10.0.1/24,fd86:ea04:1111::1/64"
```
Comment (ZeronTheXeon) [inline vps-service/restPut.go]: Same as above, if the allowedIPs are constant, can we extract them out to a global const?

**Pull request merged**